### ISSUE TITLE
[JENKINS-9104] Visual studio builds started by Jenkins fail with "Fatal error C1090" because mspdbsrv.exe gets killed

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#10814]
<p>I run into errors when using a customized build system which uses Visual Studio's devenv.exe under the hood to compile VisualStudio 2005 projects (with VC++ compiler). When starting two parallel builds with Jenkins (on different code base) the second job will always fail with "Fatal error C1090: PDB API call failed, error code '23' : '(" in exactly the same second the first job finishes processing. Running both jobs outside Jenkins does not produce the error.<br/>
This has also been reported for builds executed by MSBuild on the Jenkins user mailing list <span class="error">&#91;1&#93;</span>.</p>

<p>I analysed this issue thoroughly and can track the problem down to the usage of mspdbsrv.exe. This program is automatically spawned when building a VisualStudio project. All Visual Studio instances normally share one common pdb-server which shutdown itself after a idle period (standard is 10 minutes). "It ensures access to .pdb files is properly serialized in parallel builds when multiple instances of the compiler try to access the same .pdb file" <span class="error">&#91;2&#93;</span>.<br/>
I assume that Jenkins does a clean up of its build environment when a automatically started job finishes (like as described at <a href="http://wiki.jenkins-ci.org/display/JENKINS/Aborting+a+build" class="external-link" target="_blank" rel="nofollow noopener">http://wiki.jenkins-ci.org/display/JENKINS/Aborting+a+build</a>). I checked mspbsrv.exe with ProcessExplorer and the process indeed has a variable JENKINS_COOKIE/HUDSON_COOKIE set in its environment if started through Jenkins. Killing mspdbsrv.exe while projects are still connected will break compilation. </p>

<p>Jenkins mustn't kill mspdbsrv.exe to be able to build more than one Visual Studio project at the same time.</p>

<p>&#8211;<br/>
<span class="error">&#91;1&#93;</span> <a href="http://jenkins.361315.n4.nabble.com/MSBuild-fatal-errors-when-build-triggered-by-timer-td385181.html" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins.361315.n4.nabble.com/MSBuild-fatal-errors-when-build-triggered-by-timer-td385181.html</a><br/>
<span class="error">&#91;2&#93;</span> <a href="http://social.msdn.microsoft.com/Forums/en-US/vcgeneral/thread/b1d1bceb-06b6-47ef-a0ea-23ea752e0c4f/" class="external-link" target="_blank" rel="nofollow noopener">http://social.msdn.microsoft.com/Forums/en-US/vcgeneral/thread/b1d1bceb-06b6-47ef-a0ea-23ea752e0c4f/</a></p>

**Attachments:**

- [envinject-config.png](https://issues.jenkins.io/secure/attachment/38970/envinject-config.png)
- [screenshot.JPG](https://issues.jenkins.io/secure/attachment/32385/screenshot.JPG)

---
<details><summary><i>Originally reported by <a title="Christoph Vogtländer" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gordin">gordin</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-9104" target="_blank">Visual studio builds started by Jenkins fail with "Fatal error C1090" because mspdbsrv.exe gets killed</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Daniel Weber" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielweber">danielweber</a>
<li><b>reported by</b>: gordin
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2018-05-07T15:04:36+00:00
<li><b>votes</b>: 71
<li><b>watchers</b>: 95
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>:</i>
<p>You might refer to <a href="https://issues.jenkins.io/browse/JENKINS-3105" title="Configuration UI to disable process tree killer selectively" class="issue-link" data-issue-key="JENKINS-3105"><del>JENKINS-3105</del></a> for a work around or alternative technique.  I assume the process tree killer is what is killing the mspdbsrv.exe process, and disabling process tree killer for these jobs may avoid the problem.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Aren't you able to launch that service manually instead of having it launched by the first build to come along?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Recursive process killing on Windows was added between versions 1.16 (1.532.x) and 1.19 (1.554.x) of that library, see <a href="https://github.com/kohsuke/winp/compare/winp-1.16...winp-1.19" class="external-link" target="_blank" rel="nofollow noopener">here</a>.</p>

<p>Workaround could be to block loading of winp in some way.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gordin">gordin</a>:</i>
<p>I set the severity to minor because an easy workaround is available. What is the reason you can't use "BUILD_ID=dontKillMe" environment variable? This disables the process killer for the job (or globally if set for the whole Jenkins instance). Generally I think the process killer is a good thing, but normally it shouldn't be needed.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Christoph: Have you tried this on Windows with winp doing the killing? I think it works differently there.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gordin">gordin</a>:</i>
<p>sorry, no I haven't tried this with winp (to be honest, I don't even know what winp is). I will try with an current Jenkins setup.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Winp in the library doing the recursive killing on Windows (if available), and that is what was fixed between 1.532.x and 1.554.x &#8211; so Kevin is correct that this was changed between these versions.</p>

<p>There's a few reported issues related to winp not working reliably, maybe one of them can be exploited as a workaround to prevent it from killing pspdbsrv.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=sweavo">sweavo</a>:</i>
<p>"Aren't you able to launch that service manually instead of having it launched by the first build to come along?"</p>

<p>This is covered in the comments. The service times out, which can still happen mid-build. If you set the timeout long, then you risk memory leaks.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gordin">gordin</a>:</i>
<p>Setting the BUILD_ID to "dontKillMe" still works as expected with Jenkins 1.554.1 LTS. Even though I'm not able to test the original set up (as I don't use VS and mspdbsrv.exe any longer) a new process spawned during run with python subprocess.Popen() will not be killed by the process tree killer. Running without setting the BUILD_ID will kill the subprocess as expected.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Great! In that case, this is not a defect but behaves as intended.</p>

<p>What would be a good location to document setting BUILD_ID to prevent process killing? Obviously, there's a need there...</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gordin">gordin</a>:</i>
<p>It is already documented at <a href="https://wiki.jenkins-ci.org/display/JENKINS/ProcessTreeKiller" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/ProcessTreeKiller</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Random wiki pages aren't exactly discoverable. Unless you know it's there you wouldn't even bother searching.</p>

<p>Maybe add to the description of shell/batch build steps that launched processes are cleaned up after the script exits, and that this can be disabled?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Christoph: Shell and Batch are both in core and the most straightforward choices for launching programs. Specialist plugins might not even allow this much flexibility.</p>

<p>I'd just like to get more discoverability, and that doesn't need annotating at every conceivable location. If users knows this solution from Batch/Shell descriptions, and maybe transfer it over to similar plugin-provided builders, that's perfect.</p>

<hr />

<p>Tree killer configuration might be helpful, but that should be filed in a new issue. AFAICT this needs to touch a lot of parts, so this would be a rather large project.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>How are you starting these builds? Batch? MSBuild plugin? What exact commands? If batch, did you try setting BUILD_ID as described on <a href="https://wiki.jenkins-ci.org/display/JENKINS/ProcessTreeKiller" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/ProcessTreeKiller</a> ?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=delboyjay">delboyjay</a>:</i>
<p>Does anyone know if there is an option to stop Jenkins from killing processes completely as a global option instead of having to add the BUILD_ID to every single job? I have tried adding this as an env variable at the node level but it doesn't appear to give the desired results (same PDB errors were still occurring), maybe I'm doing something wrong or misunderstanding how this is working under the hood? </p>

<p>We were running CruiseControl for years and never had this problem but we did however have issues where processes were not terminating properly and builds would run forever until someone intervened. Sometimes we still get this with Jenkins so from my point of view one problem is better than two so I'd rather just have an option to tell Jenkins not to force terminate anything - ever. If this cannot be done with the current version (ours is 1.566) Can we at least add a check box that says "Do not auto-terminate processes" as an option in a future release and let the user decide? </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=leedega">leedega</a>:</i>
<p>One thing I felt needed to be expressed here is that the fact that this defect arose in an update to the LTS edition at all worries me. Combined with the fact that this defect has been opened and under active discussion for months now without any 'real' resolution - other than some hacks and workarounds - is even more concerning. According to the Jenkins <a href="https://wiki.jenkins-ci.org/display/JENKINS/LTS+Release+Line" class="external-link" target="_blank" rel="nofollow noopener">website</a> LTS editions should "...change(s) less often and only for important bug fixes...". This policy seems to have been completely negated here. Given the severity / impact of this change I would have expected whatever "improvement" was made that caused this problem would have been reserved for the "latest" release, or at the very least reverted from the LTS edition after this problem was discovered.</p>

<p>Perhaps someone with more knowledge about the cause of this error could elaborate on why neither of these approaches has been taken here.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kerrhome">kerrhome</a>:</i>
<p>@Del, Yes, you cannot set BUILD_ID for a slave setting.  It is set by Jenkins on a per build basis. You'd either have to set it in the batch section of the job itself (we did this for our most frequently used builds) or if you call a batch script or some other script, you can put it there.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ki82">ki82</a>:</i>
<p>200$ is up for grabs for solving this issue at: <a href="https://freedomsponsors.org/issue/596/visual-studio-builds-started-by-jenkins-fail-with-fatal-error-c1090-because-mspdbsrvexe-gets-killed" class="external-link" target="_blank" rel="nofollow noopener">https://freedomsponsors.org/issue/596/visual-studio-builds-started-by-jenkins-fail-with-fatal-error-c1090-because-mspdbsrvexe-gets-killed</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=leedega">leedega</a>:</i>
<p><b>Update</b><br/>
While reporting the issue in my last comment I had the idea for a slight variation of the configuration described there which does appear to work in both use cases. The main modification that I made was to separate the build operation into two separate build operations: </p>
<ul>
	<li>the first is a simple Windows command line call which overrides BUILD_ID and then launches mspdbsrv.exe. Once this first operation completes, Jenkins terminates the shell session that is linked to the pdbsrv process thus decoupling it from the agent. Combined with the overloaded BUILD_ID env var, Jenkins can no longer track the process.</li>
	<li>the second operation is just another instance of a Windows shell session that then calls into msbuild to proceed with the build.</li>
</ul>


<p>Theoretically even this solution "could" fall prey to the same problem I described in my previous comment, however the execution time of this initial build step is negligible and is highly unlikely to be exploited in practice (ie: a user would need to hit the kill button on the build at just that small fraction of a second it takes Jenkins to launch mspdbsrv.exe).</p>

<p>I'm not sure how easy this hack will be for us to roll out into production at the scale we need, but just in case others find this tidbit of information helpful I thought I'd provide it here.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=solstice333">solstice333</a>:</i>
<p>Just wanted to note that this also occurs on my slave nodes and each slave node only has one executor. So at first glance, since I'm not running concurrent builds on any individual slave node, it seems like this error occurring on my slave nodes doesn't make any sense.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=damiandixon">damiandixon</a>:</i>
<p>> Revert "<span class="error">&#91;FIXED JENKINS-9104&#93;</span> Veto killing mspdbsrv.exe"</p>

<p>I'm confused why has the code fix been reverted?</p>

<p>The reason I am looking at this again is that the BUILD_ID work around is no longer working for me.</p>

<p>Neither is the 1.25 msbuild plugin which is meant to have the fix in.</p>

<p>I upgraded from 1.595 to 1.645.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=damiandixon" class="user-hover" rel="damiandixon">damiandixon</a> <a href="https://github.com/jenkinsci/msbuild-plugin/pull/20" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/msbuild-plugin/pull/20</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielweber">danielweber</a>:</i>
<p>This is still not resolved. We need an update of the msbuild-plugin, see PR <a href="https://github.com/jenkinsci/msbuild-plugin/pull/21" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/msbuild-plugin/pull/21</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=akb">akb</a>:</i>
<p>Is there a plan for Visual Studio builds not started by the msbuild-plugin, please?</p>

<p>I'm asking because our job configurations use a "Execute Windows batch command" build step rather than "Build a Visual Studio project or solution using MSBuild" build step (and our batch process is non-trivial).</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=akb" class="user-hover" rel="akb">akb</a> The proposed MSBuild Plugin change only requires the plugin to be installed to be effective (assuming <tt>mspdbsrv.exe</tt> is what you don't want killed).</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>msbuild-1.26 should contain the fix. Can we finally resolve this, or is something missing?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=teljj001">teljj001</a>:</i>
<p>IMO, as soon as 1.26 is released.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mbrock">mbrock</a>:</i>
<p>Updating the msbuild plugin won't work in our situation. We run into this issue, but we don't have the plugin installed. Rather the issue comes for us in the Final Builder scripts we run via Jenkins that call msbuild.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=grillba">grillba</a>:</i>
<p>Hi Stefan, refer my comments above. This fixed it for us. Cheers</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=walteste">walteste</a>:</i>
<p>Hi <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=grillba" class="user-hover" rel="grillba">grillba</a>, thanks a lot for your suggestion. It seems that this solved our issues.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ext3h">ext3h</a>:</i>
<p>Little side note: It might not be sufficient to just specify _<em>MSPDBSRV_ENDPOINT</em>_ env variable in order to avoid conflicts. I recommend to additionally also set <em>TMP</em> , <em>TEMP</em> and <em>TEMPDIR</em> to an isolated folder if you plan on invoking MSBUILD in parallel as various plugins for MSBUILD as well as MSBUILD itself will place files there.</p>

<p>Further catch of using _<em>MSPDBSRV_ENDPOINT</em>_ is, that now serialization of parallel builds in the same working directory will break in return, unless you made sure that the tempoary files for the different architectures (e.g. the temporary program database created with the individual object files, and commonly named just e.g. "Debug\vc120.pdb", notice the lack of a prefix for the architecture) are completely isolated as well. Otherwise the different mspdbsrv-instances will now collide accessing the same file.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=billhoo">billhoo</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=grillba" class="user-hover" rel="grillba">grillba</a>, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=walteste" class="user-hover" rel="walteste">walteste</a> Hi there, we've got this issue too, and we followed your suggestions to config the master Jenkins node like this:</p>

<p>Configure system > Environment variables > Add new key value pair below:</p>

<p> </p>

<p>KEY: _<em>MSPDBSRV_ENDPOINT</em>_</p>

<p>VALUE: $BUILD_TAG</p>

<p> </p>

<p>But we got nothing, the error still raised up on windows slave, could you please explain the solution in detail? Should we set this Key-Value on the slave node? Thanks in advance</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=grillba">grillba</a>:</i>
<p>@billhoo,</p>

<p>You need to do it at the Job level - Not the system level. Use envinject to add the environment variable</p>

<p>Have a look here for how to use envinject,  <a href="https://wiki.jenkins.io/display/JENKINS/EnvInject+Plugin" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins.io/display/JENKINS/EnvInject+Plugin</a></p>

<p>Make sure you follow the "Inject variables as a build step" topic</p>

<p>Regards</p>

<p>Mark</p>

<p> </p>

<p> </p>

<p> </p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=billhoo">billhoo</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=grillba" class="user-hover" rel="grillba">grillba</a>,</p>

<p>Thanks for the timely reply, we've followed your guide and found that there were already 3 seprated mspdbsvr.exe processes(for test purpose, we've ran 3 jobs on one windows slave concurrently) ran in background, so it seems worked, but unfortunately, one of our job still failed due to C1090 error.</p>

<p> </p>

<p>This is the screenshot of EnvInject in each of our 3 Pipeline jobs configuration page,</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/38970/38970_envinject-config.png" style="border: 0px solid black" /></span></p>

<p>I don't think there's anything wrong here, do I miss something?</p>

<p> </p>

<p>Thanks,</p>

<p>Bill.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jakuborava">jakuborava</a>:</i>
<p>I had the same problem with parallel builds (eg. running in parallel job A from trunk and job A from branch), I tried the solution with _<em>MSPDBSRV_ENDPOINT</em>_ with value BUILD_TAG and it worked almost for all jobs. In one situation I still had that error. So I replaced BUILD_TAG with JOB_NAME environment variable and suddenly it was fine, for now we are out of problems. If anyone has still the problem with ENDPOINT solution, try to change BUILD_TAG for something else. If you do not allow parallel build in single job, JOB_NAME should be enough, otherwise you can try JOB_NAME + BUILD_NUMBER combination.</p>

<p>Maybe ENDPOINT has some restrictions, but I did not have a time to inspect this deeper. What I know is that the problematic job has the longest name in my Jenkins - approx. 48 characters.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Daniel Beck<br/>
Path:<br/>
 content/_data/changelogs/weekly.yml<br/>
<a href="http://jenkins-ci.org/commit/jenkins.io/0391fcb9b4c957e9e41fde03409de330a3de571d" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins.io/0391fcb9b4c957e9e41fde03409de330a3de571d</a><br/>
Log:<br/>
  Remove <a href="https://issues.jenkins.io/browse/JENKINS-9104" title="Visual studio builds started by Jenkins fail with "Fatal error C1090" because mspdbsrv.exe gets killed" class="issue-link" data-issue-key="JENKINS-9104"><del>JENKINS-9104</del></a> fix from release to unblock it</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 core/src/main/java/hudson/util/ProcessTree.java<br/>
 test/src/test/java/hudson/util/ProcessTreeKillerTest.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/3465da4764c322baf4fb5b90651ef6b9bcd409fb" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/3465da4764c322baf4fb5b90651ef6b9bcd409fb</a><br/>
Log:<br/>
  Merge pull request #3419 from dwnusbaum/<a href="https://issues.jenkins.io/browse/JENKINS-9104" title="Visual studio builds started by Jenkins fail with "Fatal error C1090" because mspdbsrv.exe gets killed" class="issue-link" data-issue-key="JENKINS-9104"><del>JENKINS-9104</del></a>-test-fix</p>

<p>Fix test failure by cleaning up static state after tests</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/ddbc4bbce7d3...3465da4764c3" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/ddbc4bbce7d3...3465da4764c3</a><br/>
      *<b>NOTE:</b>* This service been marked for deprecation: <a href="https://developer.github.com/changes/2018-04-25-github-services-deprecation/" class="external-link" target="_blank" rel="nofollow noopener">https://developer.github.com/changes/2018-04-25-github-services-deprecation/</a></p>

<p>      Functionality will be removed from GitHub.com on January 31st, 2019.</p>
